src/main/java/org/logstash/dissect/Delim.java

```diff
@@ -0,0 +1,7 @@
+package org.logstash.dissect;
+
+interface Delim {
```

Comment:
I'd say it's more usual in Java to spell out names, i.e. `Delimiter`. I also sense a mismatch between my expectation of what a delimiter represents and which methods you offer in this interface. It looks like it's just a dump holder and has no behavior. Can you encapsulate this behavior instead in the implementation(s)?